packages/@react-aria/selection/src/useSelectableCollection.ts

```diff
@@ -369,13 +373,11 @@ export function useSelectableCollection(options: AriaSelectableCollectionOptions
     let modality = getInteractionModality();
     if (manager.isFocused && manager.focusedKey != null && scrollRef?.current) {
       let element = scrollRef.current.querySelector(`[data-key="${manager.focusedKey}"]`) as HTMLElement;
-      if (element) {
+      if (element && modality === 'keyboard') {
         if (!isVirtualized) {
           scrollIntoView(scrollRef.current, element);
         }
-        if (modality === 'keyboard') {
-          scrollIntoViewport(element, {containingElement: ref.current});
-        }
+        scrollIntoViewport(element, {containingElement: ref.current});
       }
```

Comment:
The changes in this file ensure we don't scroll an item into view when clicking with a mouse, only when navigating via a keyboard. This was already the case for virtualized collections, but I guess we never made the changes here as well.